pkg/scalers/elasticsearch_scaler.go

```diff
@@ -0,0 +1,277 @@
+	if err != nil {
+		elasticsearchLog.Error(err, fmt.Sprintf("Found error when creating client: %s", err))
+		return nil, err
+	}
+
+	_, err = esClient.Info()
+	if err != nil {
+		elasticsearchLog.Error(err, fmt.Sprintf("Found error when pinging search engine: %s", err))
+		return nil, err
+	}
+	return esClient, nil
+}
+
+func (s *elasticsearchScaler) Close(ctx context.Context) error {
+	return nil
+}
+
+// IsActive returns true if there are pending messages to be processed
+func (s *elasticsearchScaler) IsActive(ctx context.Context) (bool, error) {
+	messages, err := s.getQueryResult()
+	if err != nil {
+		elasticsearchLog.Error(err, fmt.Sprintf("Error inspecting elasticsearch: %s", err))
+		return false, err
+	}
+	return messages > 0, nil
+}
+
+// getQueryResult returns result of the scaler query
+func (s *elasticsearchScaler) getQueryResult() (int, error) {
+	// Build the request body.
+	var body bytes.Buffer
+	if err := json.NewEncoder(&body).Encode(buildQuery(s.metadata)); err != nil {
+		elasticsearchLog.Error(err, "Error encoding query: %s", err)
+	}
+
+	// Run the templated search
+	res, err := s.esClient.SearchTemplate(
+		&body,
+		s.esClient.SearchTemplate.WithIndex(s.metadata.indexes...),
+	)
+	if err != nil {
+		elasticsearchLog.Error(err, fmt.Sprintf("Could not query elasticsearch: %s", err))
+		return 0, err
+	}
+
+	defer res.Body.Close()
+	b, err := ioutil.ReadAll(res.Body)
+	if err != nil {
+		return 0, err
+	}
+	v, err := getValueFromSearch(b, s.metadata.valueLocation)
+	if err != nil {
+		return 0, err
+	}
+	return v, nil
+}
+
+func buildQuery(metadata *elasticsearchMetadata) map[string]interface{} {
+	parameters := map[string]interface{}{}
+	for _, p := range metadata.parameters {
+		if p != "" {
+			kv := splitAndTrimBySep(p, ":")
+			parameters[kv[0]] = kv[1]
+		}
+	}
+	query := map[string]interface{}{
+		"id": metadata.searchTemplateName,
+	}
+	if len(parameters) > 0 {
+		query["params"] = parameters
+	}
+	return query
+}
+
+func getValueFromSearch(body []byte, valueLocation string) (int, error) {
+	r := gjson.GetBytes(body, valueLocation)
+	errorMsg := "valueLocation must point to value of type number but got: '%s'"
+	if r.Type == gjson.String {
+		q, err := strconv.Atoi(r.String())
+		if err != nil {
+			return 0, fmt.Errorf(errorMsg, r.String())
+		}
+		return q, nil
+	}
+	if r.Type != gjson.Number {
+		return 0, fmt.Errorf(errorMsg, r.Type.String())
+	}
+	return int(r.Num), nil
+}
+
+// GetMetricSpecForScaling returns the MetricSpec for the Horizontal Pod Autoscaler
+func (s *elasticsearchScaler) GetMetricSpecForScaling(context.Context) []v2beta2.MetricSpec {
+	targetValue := resource.NewQuantity(int64(s.metadata.targetValue), resource.DecimalSI)
+	metricName := kedautil.NormalizeString(fmt.Sprintf("elasticsearch-%s", s.metadata.searchTemplateName))
+
+	externalMetric := &v2beta2.ExternalMetricSource{
+		Metric: v2beta2.MetricIdentifier{
+			Name: GenerateMetricNameWithIndex(s.metadata.targetValue, metricName),
```

Comment:
This name should be generated using the ScalerIndex and the "metricName". The ScalerIndex is a parameter that all scalers receive in the constructor and which should be propagated to here in order to avoid conflicts. Also, it'd be tested during the unit test to ensure that it's correctly propagated.
[This point is explained more in depth here](https://github.com/kedacore/keda/blob/main/CREATE-NEW-SCALER.md#getmetricspecforscaling).
All the other scalers propagate it, so you can check any other :)